python/ray/serve/_private/deployment_state.py

```diff
@@ -1593,6 +1607,45 @@ def deploy(self, deployment_info: DeploymentInfo) -> bool:
         self._backoff_time_s = 1
         return True
 
+    def drop_invalid_handle_metrics(self, alive_serve_actor_ids: Set[str]) -> None:
+        """Drops handle metrics that are no longer valid.
+
+        This includes handles that live on Serve Proxy or replica actors
+        that have died and handles from which the controller hasn't
+        received an update for too long.
+        """
+
+        timeout_s = max(
+            2 * self.autoscaling_policy_manager.get_metrics_interval_s(),
+            RAY_SERVE_MIN_HANDLE_METRICS_TIMEOUT_S,
+        )
+        for handle_id, handle_metric in list(self.handle_requests.items()):
+            # Drop metrics for handles that are on Serve proxy/replica
+            # actors that have died
+            if (
+                handle_metric.handle_component
+                in [HandleComponent.PROXY, HandleComponent.REPLICA]
+                and handle_metric.actor_id not in alive_serve_actor_ids
+            ):
+                del self.handle_requests[handle_id]
+                if handle_metric.total_requests > 0:
+                    logger.info(
+                        f"Dropping metrics for handle '{handle_id}' because the Serve "
+                        f"actor it was on ({handle_metric.actor_id}) is no longer "
+                        f"alive. It had {handle_metric.total_requests} ongoing requests"
+                    )
```

Comment:
this is normal operation, so should not be an `info`-level log or we'll spam just as much as it is today

I'd suggest we make it a DEBUG log so we can flip it on if debugging issues

Reply:
Oops yeah meant to make it debug, made it info for testing